Best Practices/Challenges/UnitTests/UnitTests.Challenge/NotificationWorker.cs

```diff
@@ -0,0 +1,65 @@
+﻿using UnitTests.Challenge.Interfaces;
+using UnitTests.Challenge.Interfaces.NotificationServices;
+
+namespace UnitTests.Challenge
+{
+    /// <summary>
+    /// This part involves writing tests for part 6, follow the document for more information
+    /// </summary>
+    public class NotificationWorker
+    {
+        private const int MaxRetryAttempts = 3;
+        private const int RetryDelayMilliseconds = 1;
+
+        private readonly INotificationService _notificationService;
+
+        public NotificationWorker(INotificationService notificationService)
+        {
+            _notificationService = notificationService;
+        }
+
+        public async Task SendNotification(string message)
+        {
+            int retryCount = 0;
+            bool success = false;
+
+            while (retryCount < MaxRetryAttempts && !success)
+            {
+                try
+                {
+                    switch (_notificationService)
+                    {
+                        case IFaxNotificationService faxService:
+                            Console.WriteLine("Sending Fax");
+                            await faxService.SendNotificationAsync(message);
+                            break;
+                        case IEmailNotificationService emailService:
+                            Console.WriteLine("Sending Email");
+                            await emailService.SendNotificationAsync(message);
+                            break;
+                        case ISMSNotificationService smsService:
+                            Console.WriteLine("Sending SMS");
+                            await smsService.SendNotificationAsync(message);
+                            break;
+                        default:
+                            throw new InvalidOperationException("Unsupported notification service");
+                    }
+                    success = true;
```

Comment:
suggestion: We could use a break or return here and avoid the additional success complexity.